src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1069,3 +1077,62 @@ export const disableSmsVerificationsForUser = (
       return transformMongoError(error)
     },
   ).map(() => true)
+
+/**
+ * Checks the number of free smses sent by the admin of the form and deactivates verification or sends mail as required
+ * @param form The form whose admin's sms counts needs to be checked
+ * @returns ok(true) when the verification has been deactivated successfully or no action is required
+ * @returns err(MailGenerationError) when an error occurred on creating the HTML template for the email
+ * @returns err(MailSendError) when an error occurred on sending the email
+ * @returns err(PossibleDatabaseError) when an error occurred while retrieving the counts from the database
+ */
+export const checkFreeSmsSentByAdminAndDeactivateVerification = (
+  form: IPopulatedForm,
+): ResultAsync<
+  true,
+  MailGenerationError | MailSendError | PossibleDatabaseError
+> => {
+  if (isFormOnboarded(form)) {
+    return okAsync(true)
+  }
+
+  // Convert to string because it's typed as any
+  const formAdminId = String(form.admin._id)
+
+  return SmsService.retrieveFreeSmsCounts(formAdminId).andThen((freeSmsSent) =>
+    checkSmsCountAndPerformAction(form, freeSmsSent),
+  )
+}
+
+/**
+ * Checks the number of free smses sent by the admin of a form and performs the appropriate action
+ * @param form The form whose admin's sms counts needs to be checked
+ * @returns ok(true) when the action has been performed successfully
+ * @returns err(MailGenerationError) when an error occurred on creating the HTML template for the email
+ * @returns err(MailSendError) when an error occurred on sending the email
+ * @returns err(PossibleDatabaseError) when an error occurred while retrieving the counts from the database
+ */
+export const checkSmsCountAndPerformAction = (
```

Comment:
Doesn't need to be exported since this is a helper function

if for testing, can just test the exported function. exporting helper functions to test is a small code smell

Reply:
done [here](https://github.com/opengovsg/FormSG/pull/2326/commits/301d20ee1e4a16f06c518f6cf5ee5ff0588ca8a8)